lemur/plugins/lemur_aws/sts.py

```diff
@@ -47,17 +47,57 @@ def assume_service(account_number, service, region='us-east-1'):
             security_token=role.credentials.session_token)
 
 
+def construct_role_arn(role_type='role', **kwargs):
+    account_number = kwargs.get('account_number') or \
+        current_app.config.get('AWS_DEFAULT_ACCOUNT_NUMBER') or \
+        boto3.client('sts').get_caller_identity().get('Account')
+    arn_data = {
+        'num': account_number,
+        'role_type': role_type,
+        'name': current_app.config.get('LEMUR_INSTANCE_PROFILE', 'Lemur')
+    }
+    arn_data['service'] = 'sts' if role_type == 'assumed-role' else 'iam'
+
+    return 'arn:aws:{service}::{num}:{role_type}/{name}'.format(**arn_data)
+
+
+def already_assumed_target_role(**kwargs):
+    target_arn = construct_role_arn('assumed-role', **kwargs)
+    current_arn = boto3.client('sts').get_caller_identity().get('Arn')
+    return current_arn.startswith(target_arn)
+
+
+def _current_session_region():
+    _s = boto3._get_default_session()
+    return _s.region_name
+
+
+def _current_session_credentials_dict():
+    _c = boto3._get_default_session().get_credentials()
+    return {
+        'AccessKeyId': _c.access_key,
+        'SecretAccessKey': _c.secret_key,
+        'SessionToken': _c.token
+    }
+
+
 def sts_client(service, service_type='client'):
     def decorator(f):
         @wraps(f)
         def decorated_function(*args, **kwargs):
             sts = boto3.client('sts')
-            arn = 'arn:aws:iam::{0}:role/{1}'.format(
-                kwargs.pop('account_number'),
-                current_app.config.get('LEMUR_INSTANCE_PROFILE', 'Lemur')
-            )
-            # TODO add user specific information to RoleSessionName
-            role = sts.assume_role(RoleArn=arn, RoleSessionName='lemur')
+
+            region_name = kwargs.get('region') or _current_session_region()
+            role = None
+
+            if already_assumed_target_role(**kwargs):
+                role = {'Credentials': _current_session_credentials_dict()}
```

Comment:
I think the default behavior should remain, and only change if an `AWS_DEFAULT_ACCOUNT_NUMBER` is found. This way existing tests should pass and we can add new ones to test the presence of the modified environment.